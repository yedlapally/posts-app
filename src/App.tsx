import * as React from "react";
import "./App.scss";
import "bootstrap/dist/css/bootstrap.min.css";
import { Switch, HashRouter, Route, BrowserRouter, Redirect } from "react-router-dom";
import { IRoute } from "./model/IRoute"
import routes from "./routes/routes";

export const App = () => {
  const isUserAuthincate = () => {
    const isUserValid = window.sessionStorage.getItem("isAuthenticate");
    if (isUserValid && JSON.parse(isUserValid))
      return true;
    else return false;
  };

  const PrivateRoute: React.FC<IRoute> = ({ component: Component, ...rest }: IRoute) => (
    <Route
      {...rest}
      render={(props) =>
        isUserAuthincate() === true ? (
          <Component {...props} />
        ) : (
            <Redirect
              to={{
                pathname: "/",
              }}
            />
          )
      }
    />
  );

  return (
    <>
      <BrowserRouter>
        <Switch>
          {routes.map((route, index) =>
            route.isPrivateRoute ? (
              <PrivateRoute
                {...route}
                key={index}
              />
            ) : (
                <Route
                  path={route.path}
                  component={route.component}
                  exact={true}
                  key={index}
                />
              )
          )}
          <Route path="/" render={() => <Redirect to="/login" />} />

        </Switch>
      </BrowserRouter>
    </>
  );
};