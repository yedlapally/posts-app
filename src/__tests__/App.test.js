import React from "react";
import { shallow, mount, render, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import {App} from "../App";
import renderer from "react-test-renderer";

configure({ adapter: new Adapter() });

describe("should render PostDetailsMounted without throwing an error", () => {

  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();  
  const AppWrapper = mount(
    <App />
  );

  it("testing the app component ",()=>
  {
    expect(AppWrapper.find(".login-page").exists()).toBe(true)
  })

});