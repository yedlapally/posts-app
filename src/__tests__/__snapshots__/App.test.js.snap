// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="login-page"
>
  <div
    className="card"
  >
    <div
      className="card-header"
    >
      Login
    </div>
    <div
      className="card-body"
    >
      <form
        className="login-form"
      >
        <div
          className="form-group"
        >
          <label
            className=""
            htmlFor="exampleEmail"
          >
            Username
          </label>
          <input
            className="form-control"
            id="username"
            name="username"
            onChange={[Function]}
            placeholder="Username"
            type="email"
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <label
            className=""
            htmlFor="Password"
          >
            Password
          </label>
          <input
            className="form-control"
            id="password"
            name="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value=""
          />
        </div>
        <button
          aria-label={null}
          className="submit_btn btn btn-primary"
          onClick={[Function]}
          type="button"
          variant="contained"
        >
          Submit
        </button>
      </form>
    </div>
  </div>
</div>
`;
